src/net_processing.cpp

```diff
@@ -1282,12 +1282,18 @@ void PeerManagerImpl::Misbehaving(const NodeId pnode, const int howmuch, const s
     LOCK(peer->m_misbehavior_mutex);
     peer->m_misbehavior_score += howmuch;
     const std::string message_prefixed = message.empty() ? "" : (": " + message);
-    if (peer->m_misbehavior_score >= DISCOURAGEMENT_THRESHOLD && peer->m_misbehavior_score - howmuch < DISCOURAGEMENT_THRESHOLD) {
-        LogPrint(BCLog::NET, "Misbehaving: peer=%d (%d -> %d) DISCOURAGE THRESHOLD EXCEEDED%s\n", pnode, peer->m_misbehavior_score - howmuch, peer->m_misbehavior_score, message_prefixed);
+
+    const int score_before{peer->m_misbehavior_score - howmuch};
+    const int score_now{peer->m_misbehavior_score};
+    std::string warning{""};
```

Comment:
No need to explicitly initialize this to an empty string. `std::string`'s default initializer creates and empty string:

```suggestion
    std::string warning;
```

Reply:
Thanks, done.